indicaAi/app/controllers/favorite_locals_controller.rb

```diff
@@ -0,0 +1,57 @@
+# Favorite Controller
+class FavoriteLocalsController < ApplicationController
+  # GET /favorites
+  def index
+    @favorites = FavoriteLocal.all
+    json_response(@favorites)
+  end
+
+  # Post /favorite/create
+  def create
+    @favorite = FavoriteLocal.new(favorite_params)
+    if @favorite.valid?
+      @favorite.save
+      render_success('SUCCESS', 'Saved favorite', @favorite)
+    else
+      render_error('ERROR', 'Favorite not saved')
+    end
+  end
+
+  # Patch /favorites/update/:id
+  def update
+    @favorite = FavoriteLocal.find(params[:id])
+    @favorite.update(favorite_params)
+
+    if @favorite.save
+      render_success('SUCCESS', 'Updated favorite', @favorite)
+    else
+      render_error('ERROR', 'Not updated favorite')
+    end
+  end
+
+  # Delete /favorites/destroy/:id
+  def destroy
+    @favorite = FavoriteLocal.find(params[:id])
+    return render_success('SUCCESS', 'Deleted favorite', @favorite) if
```

Comment:
Se ele destruir com sucesso ele retorna a mensagem de Sucesso. Mas e se ele não conseguir destruir, o que ele retorna?

Reply:
Não consigo imaginar um caso pra te exemplificar agora :/

Mesmo assim é bom colocar o else porque na medida que o software evolui novas situações podem surgir. É sempre perigoso assumir que algo nunca vai ocorre no software.

No caso dos testes você pode colocar um stub e forçar o destroy a retornar falso. Seria algo como `allow(favorite_local).to receive(:destroy).and_return(false)`. É bom dar uma olhada da [Documentação do Rspec](https://relishapp.com/rspec/rspec-mocks/docs/configuring-responses/returning-a-value).